dotenv/src/parse.rs

```diff
@@ -494,8 +501,39 @@ mod variable_substitution_tests {
             assert!(false, "Expected the first value to be parsed")
         }
 
-        if let Err(LineParse(second_value)) = &parsed_values[1] {
-            assert_eq!(second_value, &String::from(">${KEY{<"))
+        if let Err(LineParse(second_value, index)) = &parsed_values[1] {
```

Comment:
use `unwrap_err` or `expect_err` here instead of `if let`.

Reply:
```
assert_eq!(
            &parsed_values[1].expect_err("whatever"),
            LineParse(wrong_value.to_owned(), wrong_value.len() - 1),
        )
```

errors with `binary operation `==` cannot be applied to type '&errors::Error'`.

If I try to use `if let` after either of `unwrap_err` or `expect_err`:
`if let LineParse(second_value, index) = parsed_values[1].expect_err("whatever") {`
 I get the `cannot move out of borrowed content` error for `parsed_values[1]`

So not sure how to write this otherwise.

Reply:
Ah yeah, that gets more complicated. You can ignore these for now then.